### ISSUE TITLE
Quest to find Pneumatic Pnowledge suddenly "completed", but book itself is nowhere to be seen

### ISSUE THREAD

**Issue by Brroleg** [CleverRaven/Cataclysm-DDA#74731]
### Describe the bug

I was clearing some settlement from monsters, was walking down the road to next house, when suddenly quest to find book of Pneumatic Pnowledge suddenly became "completed", but i didnt have the book and i could not find it anywhere around my pc when it happened. I even searched save folder with notepad++ for said book (book_pneumatics), it found mention of this book only in quest line, not anywhere in the world. Sadly i dont have safe file of the moment this quest became suddenly completed, i assumed my pc just saw this book somewhere in line of sight and thats how quest got completed so i began to search surrounds for some time and in process saved several times rewriting the save, and now save file is far from the moment of this quest became completed both in time and in place.

So, what happened? Why quest got completed if book is nowhere to be found? Possible bug? 

### Attach save file

N/A

### Steps to reproduce

1. Play the game
2. Get the quest to find book of Pneumatic Pnowledge
3...
4. Quest suddenly completed without book anywhere to be seen even in save files 

### Expected behavior

I expected to find quest book somewhere around my pc when quest to find it suddenly marked as completed 

### Versions and configuration

2024-05-23-1642

**Comment by PatrikLundell** [comment]
No, that quest is very weird. Once your character sees it it disappears and the quest is completed, depriving you of the opportunity to keep it for yourself until you find another one, as well as causing a lot of confusion.

**Comment by LyranRenegade** [comment]
I noticed the same with another mission in my mod just to find some items..
```
    {
        "id": "HA_MISSION_INFECTED_START_FIND_ANTIVIRALS",
        "type": "mission_definition",
        "name": { "str": "Find something to mitigate your infection before you die!" },
        "goal": "MGOAL_FIND_ITEM",        
        "difficulty": 1,
        "value": 0,
        "item": "antivirals",
        "origins": [ "ORIGIN_GAME_START" ]
        
    },
    ``
    
    The mission is completed once the player can see the item somewhere.. 
    (I think, or when it's somewhere near..)

**Comment by natsirt721** [comment]
Yeah, its especially weird for what is otherwise a bog-standard fetch-quest. Shouldn't the success criterion be handing the book over to Jenny?

**Comment by PatrikLundell** [comment]
Yes, it ought to be.
